### PR TITLE
mmv: 1.01b -> 2.0

### LINKED ISSUE #116515: Update mmv to new maintained upstream
###### Checklist
New upstream at https://github.com/rrthomas/mmv

Additionally, platforms should be changed to `all`.

- [x] Checked the [nixpkgs master branch](https://github.com/NixOS/nixpkgs)

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

###### Project name
`nix search` name:

current version: 1.01b
desired version: v2.0rc3

###### Notify maintainers

maintainers:

###### Note for maintainers

Please tag this issue in your PR.

### PR DESCRIPTION
**Pull request by siraben**
###### Motivation for this change
Closes #116515

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by siraben** [comment]
Build fails

<details>
<summary>build log</summary>

```
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/q4j5szj838h1wmkb31dxf4jbcfk5471b-source
source root is source
setting SOURCE_DATE_EPOCH to timestamp 1615898705 of file source/opts.ggo
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
@nix { "action": "setPhase", "phase": "autoreconfPhase" }
autoreconfPhase
autoreconf: export WARNINGS=
autoreconf: Entering directory '.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal --force -I m4
aclocal: warning: couldn't open directory 'm4': No such file or directory
autoreconf: configure.ac: tracing
autoreconf: configure.ac: not using Libtool
autoreconf: configure.ac: not using Intltool
autoreconf: configure.ac: not using Gtkdoc
autoreconf: running: /nix/store/hm8xfxprfk4l1msb9hpjwgydilbqlvl5-autoconf-2.70/bin/autoconf --force
configure.ac:29: warning: The macro `AC_PROG_CC_C99' is obsolete.
configure.ac:29: You should run autoupdate.
./lib/autoconf/c.m4:1621: AC_PROG_CC_C99 is expanded from...
configure.ac:29: the top level
configure.ac:33: warning: The macro `AC_GNU_SOURCE' is obsolete.
configure.ac:33: You should run autoupdate.
./lib/autoconf/specific.m4:312: AC_GNU_SOURCE is expanded from...
configure.ac:33: the top level
configure.ac:33: warning: AC_COMPILE_IFELSE was called before AC_USE_SYSTEM_EXTENSIONS
./lib/autoconf/specific.m4:312: AC_GNU_SOURCE is expanded from...
configure.ac:33: the top level
configure.ac:76: warning: The macro `AC_CONFIG_HEADER' is obsolete.
configure.ac:76: You should run autoupdate.
./lib/autoconf/status.m4:719: AC_CONFIG_HEADER is expanded from...
configure.ac:76: the top level
configure.ac:77: warning: AC_OUTPUT should be used without arguments.
configure.ac:77: You should run autoupdate.
autoreconf: running: /nix/store/hm8xfxprfk4l1msb9hpjwgydilbqlvl5-autoconf-2.70/bin/autoheader --force
autoreconf: running: automake --add-missing --copy --force-missing
configure.ac:31: installing 'build-aux/ar-lib'
configure.ac:29: installing 'build-aux/compile'
configure.ac:25: installing 'build-aux/install-sh'
configure.ac:25: installing 'build-aux/missing'
Makefile.am: installing './INSTALL'
configure.ac:77: error: required file 'lib/Makefile.in' not found
Makefile.am: installing 'build-aux/depcomp'
Makefile.am:22: error: required directory ./lib does not exist
autoreconf: error: automake failed with exit status: 1
```

</details>

**Comment by SuperSandro2000** [comment]
Try removing this line https://github.com/rrthomas/mmv/blob/01c1ae89e4c6879e775198f3af1a200e6182a1ae/configure.ac#L79

**Comment by siraben** [comment]
```
./configure: line 4232: gl_EARLY: command not found
checking the archiver (ar) interface... ar
checking whether ln -s works... yes
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for wchar.h... yes
checking for minix/config.h... no
checking whether it is safe to define __EXTENSIONS__... yes
checking whether _XOPEN_SOURCE should be defined... no
./configure: line 4598: gl_INIT: command not found
./configure: line 4618: syntax error near unexpected token `warnings'
./configure: line 4618: `  gl_MANYWARN_ALL_GCC(warnings)'
```

**Comment by siraben** [comment]
Looks like I can find the dependencies required here https://github.com/rrthomas/mmv/blob/master/bootstrap.conf

File: pkgs/tools/misc/mmv/default.nix [hunk 1]
```diff
@@ -1,49 +1,28 @@
...
-  '';
+  nativeBuildInputs = [ autoreconfHook gengetopt ];
+  buildInputs = [ gnulib ];
 
   meta = {
     homepage = "http://linux.maruhn.com/sec/mmv.html";
     description = "Utility for wildcard renaming, copying, etc";
-    license = lib.licenses.gpl2;
-    platforms = lib.platforms.linux;
+    license = lib.licenses.gpl3Plus;
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/mmv/default.nix:25]:
```
warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?
```

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 116518` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>mmv</li>
  </ul>
</details>

File: pkgs/tools/misc/mmv/default.nix [hunk 2]
```diff
@@ -1,49 +1,30 @@
...
-  '';
-
-  postInstall = ''
-    for variant in mcp mad mln
-    do
-      ln -s mmv "$out/bin/$variant"
-      ln -s mmv.1 "$out/share/man/man1/$variant.1"
-    done
-  '';
+  nativeBuildInputs = [ gengetopt m4 git gnupg perl autoconf automake help2man pkg-config ];
```
Comment (jtojnar) [inline pkgs/tools/misc/mmv/default.nix:20]: Heh, what is gnupg even used for?

File: pkgs/tools/misc/mmv/default.nix [hunk 3]
```diff
@@ -1,49 +1,30 @@
...
+  buildInputs = [ boehmgc ];
 
   meta = {
     homepage = "http://linux.maruhn.com/sec/mmv.html";
```
Comment (jtojnar) [inline pkgs/tools/misc/mmv/default.nix:24]: The website seems to be dead.

Comment (siraben) [reply pkgs/tools/misc/mmv/default.nix:24]: Should I point to archive.org version, or GitHub?

File: pkgs/tools/misc/mmv/default.nix (hunk 2 above)
Comment (siraben) [reply pkgs/tools/misc/mmv/default.nix:20]: I wish I knew, the bootstrap script fails if it's not detected.

```
# Build prerequisites
buildreq='
        git        1.5.5   https://git-scm.com
        gpg        1.4.11  https://www.gnupg.org
        gengetopt  2.23    https://www.gnu.org/s/gengetopt
        help2man   1.29    https://www.gnu.org/s/help2man
        perl       5.5     https://perl.com
        tar        -       https://www.gnu.org/s/tar
'
```

File: pkgs/tools/misc/mmv/default.nix (hunk 3 above)
Comment (jtojnar) [reply pkgs/tools/misc/mmv/default.nix:24]: GitHub is probably better target since it is most up to date.

**Review by jtojnar — APPROVED** [review]

**Comment by siraben** [comment]
@jtojnar merge?

**Pull request merged**